plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -1030,6 +1030,17 @@ protected void verifyTableName(DatabaseMetaData databaseMetadata, String tableNa
         }
     }
 
+    @Override
+    protected void verifyColumnName(DatabaseMetaData databaseMetadata, String columnName)
+            throws SQLException
+    {
+        // PostgreSQL truncates table name to 63 chars silently
+        // PostgreSQL driver caches the max column name length in a DatabaseMetaData object. The cost to call this method per column is low.
+        if (columnName.length() > databaseMetadata.getMaxColumnNameLength()) {
+            throw new TrinoException(NOT_SUPPORTED, format("Column name must be shorter than or equal to '%s' characters but got '%s'", databaseMetadata.getMaxColumnNameLength(), columnName.length()));
```

Comment:
Quote the contents of the `columnName` in the message besides their length.

It's important for CREATE TABLE statement, where there can be multiple potentially problematic columns in play